wp-parsely.php

```diff
@@ -21,6 +21,189 @@
 	'3.1',
 ];
 
+/**
+ * Class to hold Parse.ly loading info
+ * Will prevent this information from being queried multiple times
+ * by keeping the status in a static variable
+ */
+final class Parsely_Loader_Info {
+
+	/**
+	 * Strings for Parse.ly integration types.
+	 */
+	const INTEGRATION_TYPE_MUPLUGINS        = 'MUPLUGINS';
+	const INTEGRATION_TYPE_MUPLUGINS_SILENT = 'MUPLUGINS_SILENT';
+	const INTEGRATION_TYPE_SELF_MANAGED     = 'SELF_MANAGED';
+	const INTEGRATION_TYPE_NONE             = 'NONE';
+
+	/**
+	 * String for when no wp-parsely version can be detected.
+	 */
+	const VERSION_NONE = 'NONE';
+
+	/**
+	 * Strings for Parse.ly service types.
+	 */
+	const SERVICE_TYPE_PAID = 'PAID';
+	const SERVICE_TYPE_NONE = 'NONE';
+
+	/**
+	 * @var boolean
+	 */
+	private static $active;
+
+	/**
+	 * @var string
+	 */
+	private static $integration_type;
+
+	/**
+	 * @var array The Parse.ly WordPress options dictionary.
+	 */
+	private static $parsely;
+
+	/**
+	 * @var string
+	 */
+	private static $service_type;
+
+	/**
+	 * @var string
+	 */
+	private static $version;
+
+	/**
+	 * fetches the active status.
+	 */
+	public static function get_active() {
+		if ( null === self::$active ) {
+			self::set_active();
+		}
+
+		return self::$active;
+	}
+
+	/**
+	 * populates the private $active property if it hasn't been set.
+	 */
+	private static function set_active() {
+		$integration_type = self::get_integration_type();
+
+		if ( self::INTEGRATION_TYPE_MUPLUGINS === $integration_type ||
+			self::INTEGRATION_TYPE_MUPLUGINS_SILENT === $integration_type ||
+			self::INTEGRATION_TYPE_SELF_MANAGED === $integration_type
+		) {
+			self::$active = true;
+		}
+
+		self::$active = false;
+	}
+
+	/**
+	 * fetches the integration type.
+	 */
+	public static function get_integration_type() {
+		if ( null === self::$integration_type ) {
+			self::set_integration_type();
+		}
+
+		return self::$integration_type;
+	}
+
+	/**
+	 * populates the private $integration_type property if it hasn't been set.
+	 */
+	private static function set_integration_type() {
+		// detect a hidden installation
+		if ( get_option( '_wpvip_parsely_mu', false ) && ! has_filter( 'wpvip_parsely_load_mu' ) ) {
+			self::$integration_type = self::INTEGRATION_TYPE_MUPLUGINS_SILENT;
+		}
+
+		// mu-plugins
+		if ( has_filter( 'wpvip_parsely_load_mu' ) ) {
```

Comment:
What happens here if the `wpvip_parsely_load_mu` filter is set to `false`, and would there be any unwanted side-effects to the `set_active()` function? Is there a fix that must be applied here or in the `set_active()` function?

Reply:
I would think there is, but a quick search doesn't reveal anything. I can look into that more tomorrow if needed.

Reply:
Maybe what we could do (and maybe what @mjangda was suggesting?) is to change the `Parsely_Loader_Info` state from within the `maybe_load_plugin()` and `maybe_disable_some_features()` functions where the actual operations take place? @jesse-greathouse, can you give it a shot and see if this approach can work?

If the answer is yes, I can help you out more if needed.

Reply:
@acicovic,

The latest commit is trying to use this suggestion and I think it works well. The commit notes are as follows:

```
Changed Parsely_Loader_Info setter methods to public
Gave Parsely_Loader_Info setter methods a parameter to pass the value
Integreated Parsely_Loader_Info with maybe_load_plugin and maybe_disable_some_features methods in wp-parsely.php
Added appropriate docblock metadata for the new parameters in setters
Removed unnecessary code in Parsely_Loader_Info methods
```

If you do not mind, please look over the way I used the setter methods in `maybe_load_plugin` and `maybe_disable_some_features` to be sure that it is correctly setting the appropriate values. If I messed it up, it should be easy to fix.